cartography/intel/ontology/utils.py

```diff
@@ -0,0 +1,118 @@
+import json
+import logging
+import pkgutil
+from typing import Any
+
+import neo4j
+
+from cartography.graph.job import GraphJob
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+def load_ontology_mapping(module_name: str) -> dict:
+    """Load the ontology mapping for a given module.
+
+    This function loads the ontology mapping from a JSON file located in the
+    `cartography.data.ontology` package. The mapping file should be named
+    `<module_name>.json`, where `<module_name>` is the name of the module for
+    which the mapping is being loaded.
+
+    Args:
+        module_name (str): The name of the module for which to load the ontology mapping.
+
+    Raises:
+        ValueError: If the mapping file for the specified module is not found.
+
+    Returns:
+        dict: A dictionary containing the ontology mapping for the specified module.
+    """
+    data = pkgutil.get_data("cartography.data.ontology", f"{module_name}.json")
+    if data is None:
+        raise ValueError(f"Mapping file for {module_name} not found.")
+    return json.loads(data.decode("utf-8"))
+
+
+@timeit
+def get_source_nodes_from_graph(
+    neo4j_session: neo4j.Session,
+    source_of_truth: list[str],
+    module_name: str,
+) -> list[dict[str, Any]]:
+    """Retrieve source nodes from the Neo4j graph database based on the ontology mapping.
+
+    This function queries the Neo4j database for nodes that match the labels
+    defined in the ontology mapping for the specified module and source of truth.
+    It returns a list of dictionaries containing the relevant fields for each node.
+
+    If no source of truth is provided, default to all sources defined in the mapping.
+
+    Args:
+        neo4j_session (neo4j.Session): The Neo4j session to use for querying the database.
+        source_of_truth (list[str]): A list of source of truth identifiers to filter the modules.
+        module_name (str): The name of the ontology module to use for the mapping (eg. users, devices, etc.).
+
+    Returns:
+        list[dict[str, Any]]: A list of dictionaries, each containing a node details formatted according to the ontology mapping.
+    """
+    results: list[dict[str, Any]] = []
+    modules_mapping = load_ontology_mapping(module_name)
```

Comment:
Echoing an earlier comment in one of the json files, it'd be nice to have this be a typed object